ext/RMagick/rmimage.c

```diff
@@ -12,6 +12,22 @@
 
 #include "rmagick.h"
 
+#define SET_CHANNEL_MASK(image, channels) \
+  { \
```

Comment:
There seems to be a restriction on the order of using this macro by `{`.
If you use it for variable declaration, it will not be necessary because we have used `c99` at compiling.

Reply:
The order is important but the `{` is also there in case you use the macro twice in the same method. Without the `{` the `channel_mask` will be declared twice in the same block. That could also be avoided in the place where you use it but I would prefer to use this notation instead.

Reply:
I want to introduce `BEGIN_CHANNEL_MASK ` and `END_CHANNEL_MASK` if you like.
I guess it is easy to understand

```diff
diff --git a/ext/RMagick/rmimage.c b/ext/RMagick/rmimage.c
index 3eebb9cb..02709e2d 100644
--- a/ext/RMagick/rmimage.c
+++ b/ext/RMagick/rmimage.c
@@ -12,21 +12,25 @@
 
 #include "rmagick.h"
 
-#define SET_CHANNEL_MASK(image, channels) \
+#define BEGIN_CHANNEL_MASK \
   { \
     ChannelType \
-      channel_mask; \
+      channel_mask;
+
+#define END_CHANNEL_MASK \
+  }
+
+#define SET_CHANNEL_MASK(image, channels) \
     channel_mask=SetPixelChannelMask(image, (ChannelType)channels)
 
 #define RESTORE_CHANNEL_MASK(image) \
-    SetPixelChannelMask(image, channel_mask); \
-  }
+    SetPixelChannelMask(image, channel_mask);
 
 #define RESTORE_CHANNEL_MASKS(image, result) \
     SetPixelChannelMask(image, channel_mask); \
     if (result != (Image *)NULL) \
-      SetPixelChannelMask(result, channel_mask); \
-  }
+      SetPixelChannelMask(result, channel_mask);
+
 
 /** Method that effects an image */
 typedef Image *(effector_t)(const Image *, const double, const double, ExceptionInfo *);
```

Reply:
Not that happy with those extra macros. This would mean that we need a lot of extra macros throughout the code. And you will always do a `SET_CHANNEL_MASK` followed by `RESTORE_CHANNEL_MASK` or `RESTORE_CHANNEL_MASKS`. Those new ones will not make it clearer.

Reply:
What about the macro like below?

```c
#define WITH_CHANNEL_MASK(image, channels, expression) \
  { \
    ChannelType \
      channel_mask; \
    channel_mask=SetPixelChannelMask(image, (ChannelType)channels); \
    expression; \
    SetPixelChannelMask(image, channel_mask); \
  }

....

    WITH_CHANNEL_MASK(image, channels, {
        new_image = (fp)(image, radius, sigma, exception);
    });
```

Reply:
The expression can be quite long and I am not sure if that will improve the readability. Maybe we could rename `SET_CHANNEL_MASK` to `BEGIN_CHANNEL_MASK` and `RESTORE_CHANNEL_MASK` to `END_CHANNEL_MASK` instead? Not sure what would be a proper name for `RESTORE_CHANNEL_MASKS`. Maybe that should be a separate macro instead?

Reply:
I think it is not bad :)

Reply:
Will update the PR tonight